src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -510,6 +533,82 @@ private void ChooseActions(in Entity entity, ref MicrobeAI ai, ref CompoundAbsor
         }
     }
 
+    private (Entity Entity, Vector3 Position, CompoundBag Compounds)? GetNearestRadioactiveChunk(
+        ref WorldPosition position, float maxDistance)
+    {
+        (Entity Entity, Vector3 Position, CompoundBag Compounds)? chosenChunk = null;
+        float bestFoundChunkDistance = float.MaxValue;
+
+        BuildChunksCache();
+
+        foreach (var chunk in terrainChunkDataCache)
+        {
+            if (!chunk.Compounds.Compounds.Keys.Contains(Compound.Radiation))
+            {
+                continue;
+            }
+
+            var distance = (chunk.Position - position.Position).LengthSquared();
+
+            if (distance > bestFoundChunkDistance)
+                continue;
+
+            if (distance > maxDistance)
+                continue;
+
+            chosenChunk = chunk;
+        }
+
+        return chosenChunk;
+    }
+
+    private bool RunFromNearestRadioactiveChunk(ref WorldPosition position, ref MicrobeAI ai,
+        ref MicrobeControl control)
+    {
+        var chosenChunk = GetNearestRadioactiveChunk(ref position, 500.0f);
+
+        if (chosenChunk == null)
+        {
+            return false;
+        }
+
+        var oppositeDirection = position.Position + (position.Position - chosenChunk.Value.Position);
```

Comment:
Shouldn't this be normalized and then multiplied by a certain distance? I think that this math may end up picking a point really near if the distance between the chunk and the current position gets too low.